ddspipe_core/src/cpp/monitoring/producers/StatusMonitorProducer.cpp

```diff
@@ -0,0 +1,160 @@
+// Copyright 2024 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#include <ddspipe_core/monitoring/consumers/DdsMonitorConsumer.hpp>
+#include <ddspipe_core/monitoring/consumers/StdoutMonitorConsumer.hpp>
+#include <ddspipe_core/monitoring/producers/StatusMonitorProducer.hpp>
+
+namespace eprosima {
+namespace ddspipe {
+namespace core {
+
+
+StatusMonitorProducer* StatusMonitorProducer::instance_ = nullptr;
+
+
+void StatusMonitorProducer::init_instance(
+        StatusMonitorProducer* instance)
+{
+    instance_ = instance;
+}
+
+StatusMonitorProducer* StatusMonitorProducer::get_instance()
+{
+    if (instance_ == nullptr)
+    {
+        instance_ = new StatusMonitorProducer();
+    }
+
+    return instance_;
+}
+
+void StatusMonitorProducer::init(
+        const MonitorProducerConfiguration& configuration)
+{
+    // Store whether the producer is enabled
+    enabled_ = configuration.enabled;
+
+    if (!enabled_)
+    {
+        // Don't register the consumers if the producer is not enabled
+        return;
+    }
+
+    // Store the period so it can be used by the Monitor
+    period = configuration.period;
+
+    // TODO: initialize the consumers outside of the producer so it remains agnostic of its consumers.
```

Comment:
To what parent? To the interface?

Reply:
Much better, however it would be great if `register_consumer` was part of IMonitorProducer interface. I understand the challenge with consumers being templated, but perhaps there is a nice way of achieving it.

Reply:
I have tried a few times and failed. The only work around I see is making `IMonitorProducer` a template, which I can't do.